contracts/oracle/src/contract/alarms.rs

```diff
@@ -82,40 +76,44 @@ impl MarketAlarms {
             &below,
             AddAlarms {
                 storage,
-                addr,
+                receiver,
                 above,
                 price_alarms: Self::PRICE_ALARMS,
             },
         )
     }
 
-    pub fn try_notify_alarms<BaseC>(
-        storage: &mut dyn Storage,
-        mut batch: Batch,
-        prices: &[SpotPrice],
+    pub fn try_notify_alarms<'a, BaseC>(
+        &mut self,
+        storage: &dyn Storage,
+        prices: impl Iterator<Item = BasePrice<SwapGroup, BaseC>> + 'a,
         max_count: u32, // TODO: type alias
-    ) -> Result<Response, ContractError>
+    ) -> Result<Batch, ContractError>
     where
         BaseC: Currency,
     {
-        let mut next_id = Self::MSG_ID.may_load(storage)?.unwrap_or_default();
-        let initial_id = next_id;
-
-        Self::alarms_iter::<BaseC>(storage, prices)
+        let batch = Self::alarms_iter::<BaseC>(storage, prices)
             .take(max_count.try_into()?)
-            .try_for_each(|addr| Self::schedule_alarm(&mut batch, addr?, &mut next_id))?;
-
-        Self::MSG_ID.save(storage, &next_id)?;
-
-        let processed = next_id.wrapping_sub(initial_id);
+            .try_fold(
+                Batch::default(),
+                |mut batch, receiver| -> Result<Batch, ContractError> {
+                    // TODO: get rid of the Nls dummy type argument
+                    batch.schedule_execute_wasm_reply_always::<_, Nls>(
+                        &receiver?,
+                        ExecuteAlarmMsg::PriceAlarm(),
+                        None,
+                        batch.len().try_into()?,
+                    )?;
+                    Ok(batch)
+                },
+            )?;
 
-        Ok(Response::from(batch)
-            .set_data(to_binary(&DispatchAlarmsResponse(processed.try_into()?))?))
+        Ok(batch)
     }
 
-    pub fn try_query_alarms<BaseC>(
+    pub fn try_query_alarms<'a, BaseC>(
         storage: &dyn Storage,
-        prices: &[SpotPrice],
+        prices: impl Iterator<Item = BasePrice<SwapGroup, BaseC>> + 'a,
     ) -> Result<AlarmsStatusResponse, ContractError>
```

Comment:
from a design perspective, similar to the above issue, we shall return only `bool`